### PR TITLE
feat: add `deployer` metadata for all chains

### PR DESCRIPTION
**Pull request by paulbalaji**
resolves https://github.com/hyperlane-xyz/hyperlane-registry/issues/42

- updates to latest SDK 4.1.0 which allows addition of `deployer` details to chain metadata
	- update the schema with latest changes
- add `deployer` details for all chains, both AW and PI
- add test to check presence of `deployer` field

### CODE REVIEW

**Comment by paulbalaji** [comment]
@avious00 to review deployer details for PI chains

**Comment by avious00** [comment]
@paulbalaji could you change Michalis to Toucan Protocol https://toucan.earth/

and just spelling Merkly as Merkly Labs

otherwise LGTM thanks for getting this out 🙏

File: chains/basesepolia/metadata.yaml [hunk 1]
```diff
@@ -9,6 +9,9 @@ blocks:
   estimateBlockTime: 2
   reorgPeriod: 1
 chainId: 84532
+deployer:
+  name: Michalis Kargakis
+  url: https://github.com/0xmichalis
```
Comment (paulbalaji) [inline chains/basesepolia/metadata.yaml:14]:
```suggestion
  name: Toucan Protocol
  url: https://toucan.earth
```

File: chains/zora/metadata.yaml [hunk 2]
```diff
@@ -9,6 +9,10 @@ blocks:
   estimateBlockTime: 2
   reorgPeriod: 1
 chainId: 7777777
+deployer:
+  email: merklylabs@proton.me
+  name: merklylabs
```
Comment (paulbalaji) [inline chains/zora/metadata.yaml:14]:
```suggestion
  name: Merkly Labs
```

File: chains/alfajores/metadata.yaml [hunk 3]
```diff
@@ -13,6 +13,8 @@ blocks:
   estimateBlockTime: 5
   reorgPeriod: 0
 chainId: 44787
+deployer:
+  name: Abacus Works
```
Comment (jmrossy) [inline chains/alfajores/metadata.yaml:17]: What do you think of adding a `url: https://www.hyperlane.xyz` field to these too?

**Review by jmrossy — COMMENTED** [review]
Looks good, one comment. I do wonder if we should make a non-required CI job to check if chains have deployer info. Otherwise chains for 3rd parties would likely forget to set this (the CLI doesn't prompt for this info)

File: chains/alfajores/metadata.yaml (hunk 3 above)
Comment (paulbalaji) [reply chains/alfajores/metadata.yaml:17]: Yeah this was one thing I wasn't sure about - though we could I wasn't sure if we wanted to disambiguate between Hyperlane the protocol and AW the entity

Comment (jmrossy) [reply chains/alfajores/metadata.yaml:17]: It's a good question. When we get big enough, it's likely we'll want separate AW and Hyperlane web sites but for now maybe best to just include something that links the AW name back to official Hyperlane in some way

**Review by jmrossy — APPROVED** [review]

**Pull request merged**